csharp/src/Microsoft.ML.OnnxRuntime/NamedOnnxValue.shared.cs

```diff
@@ -221,12 +224,12 @@ public IEnumerable<T> AsEnumerable<T>()
         /// both OrtValue and pinnedMemoryHandle
         /// </summary>
         /// <param name="pinnedMemoryHandle">dispose after returned OrtValus is disposed</param>
-        /// <returns>an instance of OrtValue. The lifespan of OrtValue must overlap pinnedMemoryHandle</returns>
-        internal virtual OrtValue InputToOrtValue(NodeMetadata metadata, out IDisposable memoryOwner)
+        /// <returns>The native OrtValue handle</returns>
+        internal virtual IntPtr InputToOrtValue(NodeMetadata metadata, out IDisposable memoryOwner)
```

Comment:
I agree on the renaming part.

To answer the question. This virtual function is needed to create an array of IntPtr handles to pass it to Native Run() functions either for Input or Output as below, I think the return type better reflects that.

The memoryOwner holds on the OrtValue and the underlying either native or pinned managed memory.

The whole ordeal exists because NamedOnnxValue is not a disposable so we need to create and manage lifespan of OrtValues while Run is in progress and then cleanup.

What makes things worse, is that the derived class DisposableNamedOnnxValue is disposable, in that case, it would manage the OrtValue and we need to dance in between.